src/components/modals/PrintPreminedModal.tsx

```diff
@@ -0,0 +1,159 @@
+  const [form] = useForm<{
+    beneficary: string
+    preferUnstaked: boolean
+    memo: string
+  }>()
+
+  const [currency, setCurrency] = useState<CurrencyOption>(0)
+  const [value, setValue] = useState<string>('0')
+  const [loading, setLoading] = useState<boolean>()
+
+  async function mint() {
+    if (!contracts || !projectId || !transactor || !terminal?.version) return
+
+    setLoading(true)
+
+    await form.validateFields()
+
+    let terminalContract: Contract
+    let functionName: string
+    let args: any[]
+
+    switch (terminal.version) {
+      case '1':
+        terminalContract = contracts.TerminalV1
+        functionName = 'printPreminedTickets'
+        args = [
+          projectId.toHexString(),
+          parseWad(value).toHexString(),
+          BigNumber.from(currency).toHexString(),
+          form.getFieldValue('beneficary') ?? '',
+          form.getFieldValue('memo') ?? '',
+          form.getFieldValue('preferUnstaked'),
+        ]
+        break
+      case '1.1':
+        terminalContract = contracts.TerminalV1_1
+        functionName = 'printTickets'
+        args = [
+          projectId.toHexString(),
+          parseWad(value).toHexString(),
+          form.getFieldValue('beneficary') ?? '',
+          form.getFieldValue('memo') ?? '',
+          form.getFieldValue('preferUnstaked'),
+        ]
+    }
+
+    transactor(terminalContract, functionName, args, {
+      onConfirmed: () => {
+        form.resetFields()
+        setValue('0')
+        if (onCancel) onCancel()
+      },
+      onDone: () => {
+        setLoading(false)
+      },
+    })
+  }
+
+  const projectTargetFormItemProps:
+    | { label: string; extra: string }
+    | undefined = useMemo(() => {
+    if (!terminal?.version) return
+
+    switch (terminal.version) {
+      case '1':
+        return {
+          label: 'Payment equivalent',
+          extra:
+            'The amount of tokens minted to the receiver will be calculated based on if they had paid this amount to the project in the current funding cycle.',
+        }
+      case '1.1':
+        return {
+          label: 'Token amount',
+          extra: 'The amount of tokens to mint to the receiver.',
+        }
+    }
+  }, [terminal?.version])
+
+  const erc20Issued =
+    tokenSymbol && tokenAddress && tokenAddress !== constants.AddressZero
+
+  return (
+    <Modal
+      visible={visible}
+      onOk={mint}
+      confirmLoading={loading}
+      onCancel={onCancel}
+      okText="Mint tokens"
+    >
+      <div style={{ marginBottom: 20 }}>
+        Note: Tokens can be minted manually when allowed in the current funding
+        cycle. This can be changed by the project owner for upcoming cycles.
+      </div>
+
+      <Form layout="vertical" form={form}>
+        <Form.Item
+          label="Tokens receiver"
+          name="beneficary"
+          rules={[
+            {
+              required: true,
+              validator: (rule: any, value: any) => {
```

Comment:
No biggie, but in the future, I think we can get away with not passing these validator functions any arguments if they're not being used.

Reply:
good catch, will remove. probably copy and pasted this without thinking